File: code/jvm/im-http-api/src/main/kotlin/im/api/model/output/messages/MessageCreationOutputModel.kt
package im.api.model.output.messages

import im.domain.messages.Message
import java.time.LocalDateTime

data class MessageCreationOutputModel(
    val id: Long,
    val createdAt: LocalDateTime,
    val editedAt: LocalDateTime?,
) {
    companion object {
        fun fromDomain(message: Message): MessageCreationOutputModel =
            MessageCreationOutputModel(
                id = message.id.value,
                createdAt = message.createdAt,
                editedAt = message.editedAt,
            )
    }
}

File: code/client/web/src/Dto/output/messages/MessageCreationOutputModel.ts
/**
 * Output model for message creation
 *
 * @param id The id of the message.
 * @param createdAt The date and time when the message was created.
 * @param editedAt The date and time when the message was last edited.
 *
 */
export interface MessageCreationOutputModel {
    id: number;
    createdAt: Date;
    editedAt: Date | null;
}
